### ISSUE TITLE
[JENKINS-29198] Console logs get truncated without reason

### ISSUE THREAD

**Issue by timja-bot** [timja-org/jenkins-gh-issues-poc-11-04#8616]
<p>I have a bug which looks similar to <a href="https://issues.jenkins.io/browse/JENKINS-14899" title="consoleText stops updating at 5000 lines until end of build job" class="issue-link" data-issue-key="JENKINS-14899"><del>JENKINS-14899</del></a>:<br/>
Sometimes, when a job runs for a long time without anything getting output, the jenkins page stops updating until the job finishes. Then it add the completion lines (SUCCESS or FAILURE) to the job.<br/>
The difference is that the number of line does not matter (last bug occurence was on a job that output 264 lines (+ an unknown number of non-output lines)</p>

<p>This results in a <b>truncated /console, /consoleFull and /consoleText</b>.</p>

<p>All files do end with the two lines (if it succeeds):<br/>
    Notifying upstream projects of job completion<br/>
    Finished: SUCCESS<br/>
but hundreds of lines didn't get stored by jenkins (over 10 minutes of logs for my last job!)</p>

<p>I can often see this bug when I'm running a long impala query, where the output stops before the status line (something like "Returned 0 row(s) in 1987.42s") and the rest of the console logs never get output.</p>

<p>This happens approximately one time out of 20 for the jobs I'm running myself on this Jenkins server.</p>

---
<details><summary><i>Originally reported by <a title="Namikaze Minato" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=camusensei">camusensei</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-29198" target="_blank">Console logs get truncated without reason</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Blocker
<li><b>resolution</b>: Unresolved
<li><b>imported</b>: 2025-11-05
</ul></i>
</details>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rjfenton">rjfenton</a>:</i>
<p>As of today am also seeing this on version 2.154 I don't see any success or failure messages. I also noticed that loading output for nodes in the blue ocean view seems to give a red console error. (I'm guessing these are somehow loaded from the log which doesn't have the info since it's truncated.)</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hazok">hazok</a>:</i>
<p>I am seeing this issue as well on Jenkins 2.150.1 that comes from the "official" Docker image provided from DockerHub. I'm trying to debug an issue with a high level of parallelism and the full logs are truncated. There are logs showing in the BlueOcean view for the individual nodes, but some of those appear truncated as well, but that is difficult to confirm since the full console, consoleText, and consoleFull are truncated.</p>

<p>Additionally, I even got the infamous "Jenkins Devil" bug image directly in the console output a couple of times when trying to watch the logs live as they were output.</p>

<p>Please help provide some insight into what could cause this behavior if it could be a bug in the Pipeline code that can be worked around. Even if a there is a bug in our Pipeline code causing this issue, Jenkins should still show the full console output.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hazok">hazok</a>:</i>
<p>Please increase the priority of this issue since this is impossible for a developer to troubleshoot without console output to indicate what the issue is.</p>

<p>Issues causing outages in the console output seems as though they should be treated with the highest priority rather than Minor since there is currently no workaround.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hazok">hazok</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jetpen" class="user-hover" rel="jetpen">jetpen</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rjfenton" class="user-hover" rel="rjfenton">rjfenton</a> Please don't forget to vote for this issue if you are getting impacted. I seem to be the only one that has voted for it so far.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rizwan_inamdar">rizwan_inamdar</a>:</i>
<p>Please increase the priority.</p>

<p> </p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=laiph">laiph</a>:</i>
<p>Indeed. For us, this happens when we run like 6 builds in parallel. However, if we break them in two, having only 3 builds run at a time, same setup, just two sequence of 3 builds, we get full logs.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=elliot_nelson">elliot_nelson</a>:</i>
<p>Recently seen this a couple times on Jenkins 2.193.</p>

<p> </p>

<p>In our case, it's a multijob that ends like this:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">...
Finished Build : #18084 of Job : fvt-worker-5 with status : SUCCESS
Finished Build : #18084 of Job : fvt-worker-20 with status : SUCCESS
Finished Build : #18085 of Job : fvt-worker-7 with status : SUCCESS
Finished Build : #18085 of Job : fvt-worker-23 with status : SUCCESS
Finished Build : #18084 of Job : fvt-worker-8 with status : SUCCESS
Finished Build : Finished Build :  </pre>
</div></div>
<p> </p>

<p>It's clear the job finished successfully (including another dozen subjobs that should be listed but aren't), and that it also completed all the publisher steps as expected, but there's about 120 lines of log at the end there that are missing (affects /console, /consoleFull, and /consoleText).</p>

<p> </p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=camusensei">camusensei</a>:</i>
<p>I think you need to give more information on your problem which seems to be different from the one I used to have. Maybe open a new issue, and describe preciseley anything you can.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=JIRAUSER140876">JIRAUSER140876</a>:</i>
<p>I am seeing this issue right now on Jenkins 2.335!</p>

<p>There's something gone awry.</p>

<p>We use kubernetes and had been having some connection issues we solved by stopping the use of websockets to agents, but the truncated logs keep happening, developers cannot debug builds, and I cannot tell them why it's failed or why there are no logs...</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=JIRAUSER140876">JIRAUSER140876</a>:</i>
<p><tt>The only thing interesting I see in system jenkins logs is:</tt></p>

<p><tt>Mar 04, 2022 4:56:35 PM WARNING jenkins.branch.WorkspaceLocatorImpl getWorkspaceRoot</tt><tt><a href="https://issues.jenkins.io/browse/JENKINS-2111" title="removing a job (including multibranch/org folder branches/repos) does not remove the workspace" class="issue-link" data-issue-key="JENKINS-2111"><del>JENKINS-2111</del></a> path sanitization ineffective when using legacy Workspace Root Directory ‘${ITEM_ROOTDIR}/workspace’; switch to ‘${JENKINS_HOME}/workspace/${ITEM_FULL_NAME}’ as in <a href="https://issues.jenkins.io/browse/JENKINS-8446" title="&#39;builds&#39; directory in a separate location?" class="issue-link" data-issue-key="JENKINS-8446"><del>JENKINS-8446</del></a> / <a href="https://issues.jenkins.io/browse/JENKINS-21942" title="Unreliable defaulting of Jenkins.workspaceDir" class="issue-link" data-issue-key="JENKINS-21942">JENKINS-21942</a></tt></p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=JIRAUSER153600">JIRAUSER153600</a>:</i>
<p>Hello,</p>

<p>I'm affected by this issue too, it's really annoying as it forces user to attempt a local rebuild to obtain logs, which even worse, doesn't work when the issue is specific to the server...</p>

<p>I'm reproducing the issue with the following pipeline job:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-groovy">
pipeline {
    agent any

    stages {
        stage(<span class="code-quote">'Hello'</span>) {
            matrix {
                axes {
                    axis {
                        name <span class="code-quote">'blarg'</span>
                        values <span class="code-quote">'foo'</span>, <span class="code-quote">'bar'</span>, <span class="code-quote">'baz'</span>, <span class="code-quote">'qux'</span>, <span class="code-quote">'wizz'</span>
                    }
                }
                stages {
                    stage(<span class="code-quote">'Main'</span>) {
                        steps {
                            sh """
                            <span class="code-keyword">for</span> i <span class="code-keyword">in</span> \$(seq 1 ${TURNS}); <span class="code-keyword">do</span>
                                echo hello world \${i}!
                            done
                            """
                        }
                    }
                }
            }
        }
    }
}
</pre>
</div></div>
<p>When <b>TURNS</b> is 100000, all logs look fine, when set to 200000, the logs are truncated and the job takes 10 to 40 times longer to complete.<br/>
Note that when I redirect the logs to files by doing for example:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-sh">
(<span class="code-object">for</span> i <span class="code-object">in</span> \$(seq 1 ${TURNS}); <span class="code-object">do</span>
    echo hello world \${i}!
<span class="code-object">done</span>) | tee ${blarg}.log
</pre>
</div></div>
<p>Log files created are complete, so it doesn't look like a disk I/O bottleneck issue.</p>

<p> </p>

<p>Jenkins version: 2.375.1</p>
